### PR TITLE
[py2py3] Migration at level scr/python/A/B/C - slice 4

### LINKED ISSUE #10115: [py2py3] Migration at level scr/python/A/B/C - slice 4
#### Intro

Our purpose is to continue with the gradual migration. 
This is the 4th slice in which the migration has been divided into. 
We have split migration units at level `src/python/A/B/C`.

A detailed plan of the migration at this depth is available 
at the dmwm mkdocs [detailed plan page](https://cms-dmwm-test.docs.cern.ch/py2py3/plan_detailed/), 
with summary information on top.

#### Impact of the new feature

We start migrating the 4th bulk of migration units 
at src/python/A/B/C level.

```plaintext
                           name  nfiles  gradual  future_pm
 WMCore_Storage_SiteLocalConfig       1     True          3
               WMCore_REST_Test       1     True          8
        WMCore_WMConnectionBase       1     True          3
     WMQuality_REST_ServerSetup       1     True          5
             WMCore_REST_Server       1     True         38
          WMCore_WMStats_WebGui       1     True          0
        WMCore_Services_Service       1     True          3
      WMCore_Services_WMArchive       2     True         32
  WMCore_Services_WMStatsServer       1     True          3
            WMCore_Services_DDM       1     True          5
```

#### Is your feature request related to a problem? Please describe.

The end goal is to have WMCore py2 and py3 compatible.

#### Describe the solution you'd like

Run futurize on the first bulk of units 
and then apply the necessary manual changes.

### PR DESCRIPTION
**Pull request by mapellidario**
Fixes #10115 

#### Status

In development 

#### Related PRs

This PR is rebased on `master`

* Previous migration PR: #10117 (slice3, issue #10114  ). 
* Next migration PR: #10262 (slice5, issue #10116 )

#### Description

Run futurize and some manual changes on the first batch of src/python/A/B/C.

##### Notable change to `src/python/WMCore/Services/Service.py`

Simply adding `from builtins import str` breaks some unit tests in `test/python/WMCore_t/Services_t/Service_t.py`.

In particular the problems arise from the following lines combined with `from builtins import str`:

* https://github.com/dmwm/WMCore/blob/2e11c4ba1c280e448895dc5141e5903b647bca10/src/python/WMCore/Services/Service.py#L296
* https://github.com/dmwm/WMCore/blob/2e11c4ba1c280e448895dc5141e5903b647bca10/src/python/WMCore/Services/Service.py#L303

If I do not import `from builtins import str`, then everything is fine.

I checked the type of `data` in the unit tests and it is always py2 `str` (this means that it is a string composed of a sequence bytes) when it is not a dict/list, so doing `str(a)` in py2 has no effect. 
When I import `from builtins import str` it tries to convert a py2 `str` (sequence of bytes) to a py3 `str` (sequence of unicode codepoints), and in case the string contains non-ascii characters this conversion breaks. 

If I remove the conversion, then the unit test in py2 passes again. This should be a suitable solution for both py2 and py3. 

Only partially related: we should follow [futurize guide](https://python-future.org/compatible_idioms.html#file-io-with-open) to open files in a way that is compatible with both py2 and py3. [EDIT]

#### Is it backward compatible (if not, which system it affects?)

It should be. (Any possible cause for errors will we reported here)

#### External dependencies / deployment changes

Requires python-future in both py2 and py3 environments.

### CODE REVIEW

File: src/python/WMCore/Services/WMArchive/DataMap.py [hunk 1]
```diff
@@ -219,10 +222,10 @@ def convertOutput(outputList):
                 del outDict[field]
 
         for field in WMARCHIVE_CONVERT_TO_LIST:
-            if field in outDict and isinstance(outDict[field], basestring):
+            if field in outDict and (isinstance(outDict[field], str) or isinstance(outDict[field], bytes)):
```
Comment (amaltaro) [inline src/python/WMCore/Services/WMArchive/DataMap.py:225]: Same comment here, you can compare against a tuple of types: isinstance(outDict[field], (str, bytes))

**Review by amaltaro — CHANGES_REQUESTED** [review]
These changes look good, Dario. There is just one minor change that can be made to the code, could you please update it? Thanks

File: src/python/WMCore/Services/WMArchive/DataMap.py (hunk 1 above)
Comment (mapellidario) [reply src/python/WMCore/Services/WMArchive/DataMap.py]: Awlays correct! I just pushed a new commit!

**Comment by amaltaro** [comment]
Thanks Dario! Can you please squash commits 1 and 3? Then it's good to go as well.

**Review by amaltaro — APPROVED** [review]

**Pull request merged**